presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1695,6 +1695,18 @@ protected Scope visitExcept(Except node, Optional<Scope> scope)
             return visitSetOperation(node, scope);
         }
 
+        private <T extends Expression> List<Expression> extractDereferenceExpressionsAndIdentifiers(T expression)
+        {
+            if (expression instanceof DereferenceExpression || expression instanceof Identifier) {
+                return ImmutableList.of(expression);
+            }
+            ImmutableList.Builder<Expression> dereferenceExpression = new ImmutableList.Builder<>();
+            for (Node node : expression.getChildren()) {
+                dereferenceExpression.addAll(extractDereferenceExpressionsAndIdentifiers((Expression) node));
+            }
+            return dereferenceExpression.build();
+        }
```

Comment:
I think there is an ExpressionUtils class? If so move this function there. This could be generally useful (or use one if already exists).